src/main/java/io/osirrc/ciff/lucene/ExportAnseriniLuceneIndex.java

```diff
@@ -102,18 +109,32 @@ public static void main(String[] argv) throws Exception {
     }
 
     IndexReader reader = DirectoryReader.open(FSDirectory.open(Paths.get(args.index)));
+
+    if (reader.leaves().size() != 1) {
+      throw new RuntimeException("There should be only one leaf, index the collection using the -optimize flag");
+    }
+
     FileOutputStream fileOut = new FileOutputStream(args.output);
 
-    // Go through the index once to count the number of postings lists we're going to export.
+    // Go through the index once to count the number of postings lists we're going to export and the vocab size.
     // We need this value to populate the header.
-    System.out.println("Taking a pass through the index to count the number of postings to export...");
-    int expectedPostingsLists = countPostingsLists(reader, args.contentsField, terms);
-    System.out.println(String.format("%d postings lists expected.", expectedPostingsLists));
+    System.out.println("Taking a pass through the index to count the number of postings in the index...");
+    Counts counts = countPostingsLists(reader, args.contentsField, terms);
+    System.out.println(String.format("Exporting %d postings lists out of %d total",
+        counts.export, counts.total));
 
     // Write the header.
     System.out.println("Writing the header...");
-    Header.newBuilder().setNumPostingsLists(expectedPostingsLists).setNumDocs(
-        reader.maxDoc()).build().writeDelimitedTo(fileOut);
+    Header.newBuilder()
+        .setVersion(1)
```

Comment:
Should we have a const version variable instead?

Reply:
Check updates.